pkg/antctl/raw/multicluster/resourceimport.go

```diff
@@ -0,0 +1,206 @@
+		return
+	}
+}
+
+func init() {
+	command = &cobra.Command{
+		Use: "resourceimport",
+		Aliases: []string{
+			"resourceimports",
+			"ri",
+		},
+		Short:   "Print Multi-Cluster ResourceImports",
+		Args:    cobra.MaximumNArgs(1),
+		Example: resourceImportExample,
+		RunE:    runE,
+	}
+	o := &resourceImportOptions{}
+	options = o
+	command.Flags().StringVarP(&o.namespace, "namespace", "n", "", "Namespace of ResourceImports")
+	command.Flags().StringVarP(&o.outputFormat, "output", "o", "", "Output format. Supported formats: json|yaml")
+	command.Flags().BoolVarP(&o.allNamespace, "all-namespaces", "A", false, "If present, list ResourceImports across all Namespaces")
+
+	GetCmd.AddCommand(command)
+}
+
+func runE(cmd *cobra.Command, args []string) error {
+	options.validateAndComplete()
+	argsNum := len(args)
+	if options.allNamespace && argsNum > 0 {
+		return fmt.Errorf("a resource cannot be retrieved by name across all Namespaces")
+	}
+
+	kubeconfig, err := raw.ResolveKubeconfig(cmd)
+	if err != nil {
+		return err
+	}
+	kubeconfig.GroupVersion = &schema.GroupVersion{Group: "", Version: ""}
+	restconfigTmpl := rest.CopyConfig(kubeconfig)
+	raw.SetupKubeconfig(restconfigTmpl)
+
+	scheme := k8sruntime.NewScheme()
+	err = mcsscheme.AddToScheme(scheme)
+	if err != nil {
+		return err
+	}
+	err = antreamcscheme.AddToScheme(scheme)
+	if err != nil {
+		return err
+	}
+	err = k8sscheme.AddToScheme(scheme)
+	if err != nil {
+		return err
+	}
+	k8sClient, err := client.New(kubeconfig, client.Options{Scheme: scheme})
+	if err != nil {
+		return err
+	}
+
+	var res []multiclusterv1alpha1.ResourceImport
+	if argsNum == 0 {
+		resourceImportList := &multiclusterv1alpha1.ResourceImportList{}
+		err = k8sClient.List(context.TODO(), resourceImportList, &client.ListOptions{Namespace: options.namespace})
+		if err != nil {
+			return err
+		}
+
+		if len(resourceImportList.Items) == 0 {
+			if options.namespace != "" {
+				fmt.Printf("No resource found in Namespace %s\n", options.namespace)
+			} else {
+				fmt.Println("No resource found in all Namespaces")
+			}
+			return nil
+		}
+
+		for _, ri := range resourceImportList.Items {
+			res = append(res, ri)
+		}
+	} else {
+		resourceImportName := args[0]
+		resourceImport := multiclusterv1alpha1.ResourceImport{}
+		err = k8sClient.Get(context.TODO(), types.NamespacedName{
+			Namespace: options.namespace,
+			Name:      resourceImportName,
+		}, &resourceImport)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				return fmt.Errorf("ResourceImport %s not found in Namespace %s", resourceImportName, options.namespace)
+			}
+
+			return err
+		}
+		res = append(res, resourceImport)
+	}
+
+	switch options.outputFormat {
+	case "json":
+		byteJson, err := json.Marshal(res)
+		if err != nil {
+			return err
+		}
+		var prettyJson bytes.Buffer
+		err = json.Indent(&prettyJson, byteJson, "", " ")
+		if err != nil {
+			return err
+		}
+
+		fmt.Println(prettyJson.String())
```

Comment:
you shouldn't print json in an array when user gives one specific ResourceImport.

Reply:
done